homeworks/ReactProject/src/pages/coursePage/CoursesPage.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import {Container, Row} from 'reactstrap';
+
+
+import {SearchBar} from './components/SearchBar'
+import {CourseItemList} from './components/CourseItemList'
+
+import './styles/course-page.css'
+
+export class CoursesPage extends React.Component {
+    state = {
+        courses: []
+    }
+    
+    componentWillMount() {
+        this.getCourses();
+      
+    }
+
+    getCourses = () => {
```

Comment:
This is a side-effect in presentation component. It will cause to impossibility to reuse this component.
Move this to the HOC or container component.
https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0
https://github.com/acdlite/recompose/blob/master/docs/API.md#lifecycle

Reply:
get courses was relocated in hoc